src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -970,7 +971,9 @@ class ProtectedPathSpec extends HealthCheckSpecification {
         then: "Flow state is changed to DOWN"
         Wrappers.wait(WAIT_OFFSET) {
             assert northbound.getFlowStatus(flow.id).status == FlowState.DOWN
-            assert northbound.getFlowHistory(flow.id).last().payload.find { it.action == REROUTE_FAIL }
+            assert northbound.getFlowHistory(flow.id).find {
+                it.action == REROUTE_ACTION && it.taskId =~ (/.+ : retry #1 ignore_bw true/)
+            }
```

Comment:
This is a questionable solution, since we are not checking that the reroute operation has finished. You only check that flow is down (can be the result of original reroute or initial status change before reroute) and that the reroute retry has happened. You are not checking that reroute action has finished. Previous implementation did check the final 'REROUTE_FAIL' status of the operation.
same about other similar places in this pr

Reply:
fixed